crates/ide-db/src/path_transform.rs

```diff
@@ -326,13 +326,34 @@ impl Ctx<'_> {
             }
             result
         }
+        fn find_child_ident_pats(root_path: &SyntaxNode) -> Vec<ast::IdentPat> {
+            let mut result = Vec::new();
+            for child in root_path.children() {
```

Comment:
Maybe it'll be better to do one pass and have a `Vec` of `Either<ast::Path, ast::IdentPat>`, but I'll leave that to your judgment.